src/wodan/wodan.mli

```diff
@@ -134,36 +141,64 @@ module type S = sig
   val string_of_value : value -> string
 
   val next_key : key -> key
+  (** The next highest key
+
+      Raises Invalid_argument if already at the highest possible key *)
 
   val is_tombstone : root -> value -> bool
+  (** Whether a value is a tombstone within a root *)
 
   val insert : root -> key -> value -> unit Lwt.t
+  (** Store data in the filesystem
+
+      Any previously stored value will be silently overwritten *)
 
   val lookup : root -> key -> value option Lwt.t
+  (** Read data from the filesystem *)
 
   val mem : root -> key -> bool Lwt.t
+  (** Check whether a key has been set within the filesystem *)
 
   val flush : root -> int64 Lwt.t
+  (** Send changes to disk *)
 
   val fstrim : root -> int64 Lwt.t
+  (** Discard all blocks which the filesystem doesn't explicitly use *)
 
   val live_trim : root -> int64 Lwt.t
+  (** Discard blocks that have been unused since mounting
+   or since the last live_trim call *)
 
   val log_statistics : root -> unit
+  (** Send statistics about operations to the log *)
 
   val search_range :
     root -> key -> key -> (key -> value -> unit) -> unit Lwt.t
+  (** Call back a function for all elements in the range from start inclusive to end_ exclusive
+
+      Results are in no particular order. *)
 
   val iter : root -> (key -> value -> unit) -> unit Lwt.t
+  (** Call back a function for all elements in the filesystem *)
 
   val prepare_io :
     deviceOpenMode -> disk -> mount_options -> (root * int64) Lwt.t
+  (** Open a filesystem
+
+      Returns a root and its generation number.
+      When integrating Wodan as part of a distributed system,
+      you may want to check here that the generation number
+      has grown since the last flush *)
 end
 
+(** Build a Wodan.S module given a backing device and parameters
+
+    This is the main entry point to Wodan. *)
```

Comment:
This should be after the module declaration, but ocamlformat cant see it, reported in https://github.com/ocaml-ppx/ocamlformat/issues/723